### PR TITLE
feat(*): add `_DISTRO` arrays/vars

### PR DESCRIPTION
**Pull request by oklopfer**
## Purpose

`array_arm64`
`array_ubuntu`
`array_jammy`
`array_ubuntu_arm64`
`array_jammy_arm64`

## Approach

https://github.com/pacstall/pacstall-programs/blob/09adc2d151bcdbf010e766bb61d8e950e9167bb6/scripts/srcinfo.sh#L39 + https://github.com/pacstall/pacup/blob/72aa2fb1e173b17fb4fea0e7b2d9ff2026d4acdc/scripts/srcinfo.sh#L39 need to be changed to 
`distros=$(awk -v ORS=' ' '{sub(/\/.*/, "", $1); gsub(/:$/, "", $1); print $1}' distrolist)`

## Progress

- [x] do some
- [x] make checks
- [x] test it

## Checklist

- [x] I confirm that I have read the [contributing guidelines](https://github.com/pacstall/pacstall/blob/develop/CONTRIBUTING.md), and this pull request is abiding by all the clauses stated in the guideline.

### CODE REVIEW

File: misc/scripts/checks.sh [hunk 1]
```diff
@@ -318,9 +345,14 @@ function lint_relations() {
 }
 
 function lint_hash() {
-    local ret=0 test_hash harch test_hashsum_type test_hashsum_style test_hash_arch test_hashsum_method test_hashsum_value
-    local known_archs_hash=("amd64" "arm64" "armel" "armhf" "i386" "mips64el" "ppc64el" "riscv64" "s390x")
-    local test_hashsums=("b2" "sha512" "sha384" "sha256" "sha224" "sha1" "md5")
+    local ret=0 test_hash harch test_hashsum_type test_hashsum_style test_hash_arch test_hashsum_method test_hashsum_value \
+        hash_distro_archs known_archs_hash=("amd64" "arm64" "armel" "armhf" "i386" "mips64el" "ppc64el" "riscv64" "s390x") \
```
Comment (Elsie19) [inline misc/scripts/checks.sh:349]: I don't like how many times this is duplicated.

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: misc/scripts/checks.sh (hunk 1 above)
Comment (oklopfer) [reply misc/scripts/checks.sh:349]: what is "this"?

File: misc/scripts/checks.sh [hunk 2]
```diff
@@ -225,14 +234,23 @@ function lint_pipe_check() {
 }
 
 function lint_deps() {
-    local dep_type dep_array ret=0 dep idx kdarch known_archs_deps=("amd64" "arm64" "armel" "armhf" "i386" "mips64el" "ppc64el" "riscv64" "s390x")
+    local dep_type dep_array ret=0 dep idx kdarch kdistro kddarch \
+        known_archs_deps=("amd64" "arm64" "armel" "armhf" "i386" "mips64el" "ppc64el" "riscv64" "s390x")
```
Comment (Elsie19) [inline misc/scripts/checks.sh:238]: One.

File: misc/scripts/checks.sh [hunk 3]
```diff
@@ -292,14 +310,23 @@ function lint_ppa() {
 }
 
 function lint_relations() {
-    local rel_type rel_array ret=0 rela idx rdarch known_archs_rel=("amd64" "arm64" "armel" "armhf" "i386" "mips64el" "ppc64el" "riscv64" "s390x")
+    local rel_type rel_array ret=0 rela idx rdarch rdistro rddarch \
+        known_archs_rel=("amd64" "arm64" "armel" "armhf" "i386" "mips64el" "ppc64el" "riscv64" "s390x")
```
Comment (Elsie19) [inline misc/scripts/checks.sh:314]: Two.

**Review by Elsie19 — APPROVED** [review]

**Pull request merged**